packages/data/src/navigation/reducer.js

```diff
@@ -29,6 +32,100 @@ const reducer = (
 				menuItems: [ ...state.menuItems, ...menuItems ],
 			};
 			break;
+		case TYPES.GET_FAVORITES_FAILURE:
+			state = {
+				...state,
+				requesting: {
+					...state.requesting,
+					getFavorites: false,
+				},
+			};
+			break;
+		case TYPES.GET_FAVORITES_REQUEST:
+			state = {
+				...state,
+				requesting: {
+					...state.requesting,
+					getFavorites: true,
+				},
+			};
+			break;
+		case TYPES.GET_FAVORITES_SUCCESS:
+			state = {
+				...state,
+				favorites,
+				requesting: {
+					...state.requesting,
+					getFavorites: false,
+				},
+			};
+			break;
+		case TYPES.ADD_FAVORITE_FAILURE:
+			state = {
+				...state,
+				error,
+				requesting: {
+					...state.requesting,
+					addTodo: false,
+				},
+			};
+			break;
+		case TYPES.ADD_FAVORITE_REQUEST:
+			state = {
+				...state,
+				requesting: {
+					...state.requesting,
+					addTodo: true,
+				},
+			};
+			break;
+		case TYPES.ADD_FAVORITE_SUCCESS:
+			const newFavorites = ! state.favorites.includes( favorite )
+				? [ ...state.favorites, favorite ]
+				: state.favorites;
+
+			state = {
+				...state,
+				favorites: newFavorites,
+				requesting: {
+					...state.requesting,
+					addTodo: false,
+				},
+			};
+			break;
+		case TYPES.REMOVE_FAVORITE_FAILURE:
+			state = {
+				...state,
+				requesting: {
+					...state.requesting,
+					error,
+					removeFavorite: false,
+				},
+			};
+			break;
+		case TYPES.REMOVE_FAVORITE_REQUEST:
+			state = {
+				...state,
+				requesting: {
+					...state.requesting,
+					removeFavorite: true,
+				},
+			};
+			break;
+		case TYPES.REMOVE_FAVORITE_SUCCESS:
+			const filteredFavorites = state.favorites.filter(
```

Comment:
Currently returning the revised (remaining) favorites from this request, so we could just use the response from the API here.